src/thread_freezer.cpp

```diff
@@ -16,28 +16,36 @@ NtGetNextThread(HANDLE ProcessHandle, HANDLE ThreadHandle, ACCESS_MASK DesiredAc
 
 namespace safetyhook {
 ThreadFreezer::ThreadFreezer() {
+    m_frozen_threads.reserve(1024);
```

Comment:
Is it possible we would need more than 1024 reserved?

Reply:
> Is it possible we would need more than 1024 reserved?

It's not ideal. I think @cursey has some other ideas on how to approach this but this was a quick and easy solution. The allocations done here are the issue I believe. Cursey would know more about it though since he's the one looking into it.

Reply:
The problem was under certain circumstances the default C++ allocator would call `new` which would randomly deadlock during thread freezing (only ever saw this with debug builds but it may have been an issue in release as well).

Pre-allocating was just a quick fix solution I came up with @ 5am. I've pushed a commit that removes this and just uses a custom allocator where I am no longer running into the deadlock.